main.go

```diff
@@ -96,6 +96,7 @@ func main() {
 			log.Fatalf("Leader checker returned the following error: %s", err)
 		}
 		wg.Done()
+		panic("Leader checker process fail")
```

Comment:
Why would we need this?

Reply:
Process might receive signal to stop. For example, from context or from user. In this way it should exit gracefully without panic

Reply:
We already have check for error and `log.Fatalf()` exactly for that case, no?